### PR TITLE
Add persistence with redis

### PR DESCRIPTION
**Pull request by arreche**
This is a spike to explore a way to provide a minimum amount of reliability during voting sessions.

Currently, voting sessions cannot be resumed because all data is stored in a volatile memory.

Data loss usually happens when...
- Service provider sleeps containers.  (i.e. Heroku, Drie, Now...)
- The application is re-deployed. (Hotfixes, new features...)
- On any other unexpected error.

The proposed solution persist the voting session in an external Redis instance so it can be restored at any time.

Main changes
- Added Redis dependency with a environment variable to set up the database URL.
- Replace on memory Javascript `Map` by Redis `Strings` to store/retrieve data formated in `JSON`.
- Extracted some code from the `/commands` and `/actions` endpoints into new functions `formatStart`, `formatResult` and `formatRegister` in order to imporove the readability.
- Added new test to verify that data is persisted propertly.

### CODE REVIEW

File: async-voter.js [hunk 1]
```diff
@@ -3,156 +3,168 @@ const request = require('request')
...
-    app.get('/oauth', (req, res) => {
-        if (!req.query.code) {
-            res.status(500)
-            res.send({"Error": "Looks like we're not getting code."})
+  })
+
+  app.get('/oauth', (req, res) => {
+    if (!req.query.code) {
+      res.status(500)
+      res.send({'Error': "Looks like we're not getting code."})
```
Comment (Yorkshireman) [inline async-voter.js]: Could you make this error message more semantic? I'm not sure what it means.

Comment (Yorkshireman) [inline async-voter.js]: I know it's not part of the PR though, so I guess you could leave it - up to you I don't mind.

**Comment by arreche** [comment]
Hi, @Yorkshireman! I am super excited for having you here reviewing this code!

Please don't apologize! It was my fault. I am going to add some context in the description of this PR.

File: async-voter.js (hunk 1 above)
Comment (arreche) [inline async-voter.js]: @Yorkshireman  Good catch! Would it be possible to open a new issue about improving the user experience during the app installation?

**Pull request merged**